scripts/stubsabot.py

```diff
@@ -349,34 +350,48 @@ async def get_github_repo_info(session: aiohttp.ClientSession, stub_info: StubMe
             github_tags_info_url = f"https://api.github.com/repos/{url_path}/tags"
             async with session.get(github_tags_info_url, headers=get_github_api_headers()) as response:
                 if response.status == HTTPStatus.OK:
-                    tags: list[dict[str, Any]] = await response.json()
-                    assert isinstance(tags, list)
-                    return GitHubInfo(repo_path=url_path, tags=tags)
+                    tags = [tag["name"] for tag in await response.json()]
+                    return GitHostInfo(host="github", repo_path=url_path, tags=tags)
+        if split_url.netloc == "gitlab.com":
+            url_path = split_url.path.strip("/")
+            assert len(Path(url_path).parts) == 2
+            gitlab_tags_info_url = f"https://gitlab.com/api/v4/projects/{url_path.replace('/', '%2F')}/repository/tags"
+            async with session.get(gitlab_tags_info_url) as response:
+                if response.status == HTTPStatus.OK:
+                    tags = [tag["name"] for tag in await response.json()]
+                    return GitHostInfo(host="gitlab", repo_path=url_path, tags=tags)
```

Comment:
This can also be shortened by using one construction to avoid repeating the code, something like this:
```suggestion
        host = split_url.netloc.split('.')[-2]
        if host in ("github", "gitlab"):
            url_path = split_url.path.strip("/")
            assert len(Path(url_path).parts) == 2
            if host == "github":
                githost_tags_info_url = f"https://api.github.com/repos/{url_path}/tags"
                headers = get_github_api_headers()
            else:
                githost_tags_info_url = f"https://gitlab.com/api/v4/projects/{url_path.replace('/', '%2F')}/repository/tags"
                headers = None
            async with session.get(githost_tags_info_url, headers=headers) as response:
                if response.status == HTTPStatus.OK:
                    tags = [tag["name"] for tag in await response.json()]
                    return GitHostInfo(host=host, repo_path=url_path, tags=tags)
```